File: Commons/Action/src/main/java/com/greentree/commons/action/observer/object/EventAction.java
package com.greentree.commons.action.observer.object;

import java.util.function.Consumer;

import com.greentree.commons.action.MultiAction;


/**
 * @author Arseny Latyshev
 */
public class EventAction<T> extends MultiAction<Consumer<? super T>> implements IObjectAction<T> {
	private static final long serialVersionUID = 1L;

	@Override
	public void event(T t) {
		for(var l : listeners)
			l.accept(t);
	}



}

File: Commons/Action/src/main/java/com/greentree/commons/action/observer/object/EventAction.kt
package com.greentree.commons.action.observer.`object`

import com.greentree.commons.action.MultiAction

/**
 * @author Arseny Latyshev
 */
open class EventAction<T> : MultiAction<(T) -> Unit>(), IObjectAction<T> {

	override val listenerCount: Int
		get() = listeners.size()

	override fun event(t: T) {
		for(l in listeners)
			l(t)
	}
}
